### PR TITLE
Add TriggerValidate from Event Listener to test builder

### PR DESCRIPTION
**Pull request by khrm**
# Changes
Add TriggerValidate from Event Listener to test builder

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
```

### CODE REVIEW

**Comment by khrm** [comment]
@vtereso Added Test builder for validation.

File: test/builder/eventlistener.go [hunk 1]
```diff
@@ -106,3 +109,47 @@ func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v
 
 	return t
 }
+
+// EventListenerTriggerValidate adds a TrjggerValidate to the Triger in EventListenerSpec Triggers.
```
Comment (ncskier) [inline test/builder/eventlistener.go]: typo: `TrjggerValidate` -> `TriggerValidate` 🙂

File: test/builder/eventlistener.go [hunk 2]
```diff
@@ -106,3 +109,47 @@ func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v
...
+func EventListenerTriggerValidate(ops ...EventListenerTriggerValidateOp) EventListenerTriggerOp {
+	return func(trigger *v1alpha1.EventListenerTrigger) {
+		validate := &v1alpha1.TriggerValidate{}
+		for _, op := range ops {
+			op(validate)
+		}
+		trigger.TriggerValidate = validate
```
Comment (ncskier) [inline test/builder/eventlistener.go]: nit: the `trigger.TriggerValidate` already exists, so this code could be condensed a little with:
```
return func(trigger *v1alpha1.EventListenerTrigger) {
  for _, op := range ops {
    op(&trigger.TriggerValidate)
  }
}
```

File: test/builder/eventlistener.go [hunk 3]
```diff
@@ -106,3 +109,47 @@ func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v
...
+	return func(validate *v1alpha1.TriggerValidate) {
+		validate.TaskRef = pipelinev1.TaskRef{
+			Name:       taskName,
+			Kind:       kind,
+			APIVersion: apiVersion,
+		}
+	}
+}
+
+// EventListenerTriggerValidateServiceAccount adds a service account name to the TrigerValidate.
```
Comment (ncskier) [inline test/builder/eventlistener.go]: typo: (same as previous one) `TrigerValidate` -> `TriggerValidate`

File: test/builder/eventlistener.go [hunk 4]
```diff
@@ -106,3 +109,47 @@ func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v
...
+	return func(trigger *v1alpha1.EventListenerTrigger) {
+		validate := &v1alpha1.TriggerValidate{}
+		for _, op := range ops {
+			op(validate)
+		}
+		trigger.TriggerValidate = validate
+	}
+}
+
+// EventListenerTriggerValidateTaskRef adds a TaskRef to the TrigerValidate.
```
Comment (ncskier) [inline test/builder/eventlistener.go]: typo: `TrigerValidate` -> `TriggerValidate`

File: test/builder/eventlistener.go [hunk 5]
```diff
@@ -106,3 +109,47 @@ func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v
...
+}
+
+// EventListenerTriggerValidateServiceAccount adds a service account name to the TrigerValidate.
+func EventListenerTriggerValidateServiceAccount(serviceAccount string) EventListenerTriggerValidateOp {
+	return func(validate *v1alpha1.TriggerValidate) {
+		validate.ServiceAccountName = serviceAccount
+	}
+}
+
+// EventListenerTriggerValidateParam adds a param name to the TrigerValidate.
```
Comment (ncskier) [inline test/builder/eventlistener.go]: typo: (same as previous one) `TrigerValidate` -> `TriggerValidate`

File: test/builder/eventlistener_test.go [hunk 6]
```diff
@@ -218,6 +219,76 @@ func TestEventListenerBuilder(t *testing.T) {
...
+									Kind:       pipelinev1.NamespacedTaskKind,
+									APIVersion: "v1alpha1",
+								},
+								ServiceAccountName: "foo",
+								Params: []pipelinev1.Param{{
+									Name: "Secret",
+									Value: pipelinev1.ArrayOrString{
+										Type:      pipelinev1.ParamTypeString,
+										StringVal: "github-secret",
+									},
```
Comment (ncskier) [inline test/builder/eventlistener_test.go]: nit: I found it a little jarring when reading the code that this Param and the next one were indented differently even though they are in the same list. It's just my personal preference, but I would prefer if this Param had the same indentation as the next one:
```
Params: []pipelinev1.Param{
  {
    Name: "Secret",
    Value: pipelinev1.ArrayOrString{
      Type:      pipelinev1.ParamTypeString,
      StringVal: "github-secret",
    },
  },
...
```

File: pkg/sink/sink_test.go [hunk 7]
```diff
@@ -498,12 +498,10 @@ func TestResource_validateEvent(t *testing.T) {
 		EventListenerNamespace: "foo",
 	}
 
-	triggerValidate := &triggersv1.TriggerValidate{
-		TaskRef: pipelinev1.TaskRef{
-			Name: "bar",
-		},
-		ServiceAccountName: "foo",
-	}
+	triggerValidate := &triggersv1.TriggerValidate{}
```
Comment (vtereso) [inline pkg/sink/sink_test.go]: The top level validate builder `EventListenerTriggerValidate()` is not used here

File: pkg/sink/sink_test.go [hunk 8]
```diff
@@ -632,19 +630,10 @@ func TestResource_createValidateTask(t *testing.T) {
...
-		Params: []pipelinev1.Param{{
-			Name: "Secret",
-			Value: pipelinev1.ArrayOrString{
-				Type:      pipelinev1.ParamTypeString,
-				StringVal: "github-secret",
-			},
-		}},
-	}
+	triggerValidate := &triggersv1.TriggerValidate{}
+	bldr.EventListenerTriggerValidateTaskRef("bar", "v1alpha1", pipelinev1.NamespacedTaskKind)(triggerValidate)
```
Comment (vtereso) [inline pkg/sink/sink_test.go]: Same here

File: test/builder/eventlistener.go (hunk 4 above)
Comment (khrm) [reply test/builder/eventlistener.go]: Good catch.

File: test/builder/eventlistener.go (hunk 2 above)
Comment (ncskier) [reply test/builder/eventlistener.go]: Oh, ok. I didn't realize that `TriggerValidate` was a pointer; all of the other fields are not pointers. My mistake 😁 
Out of curiosity, why is `TriggerValidate` a pointer?

Comment (khrm) [reply test/builder/eventlistener.go]: Because it is optional. We check whether it's there or not. If it's there, we do the validation.

**Pull request merged**